app/js/actions/jobs.js

```diff
@@ -1,13 +1,94 @@
-export const createJob = (job) => ({
-    type: 'CREATE_JOB',
+import actions from '../actions';
+import { makeB64Digest } from '../util/auth';
+
+// global variable to set and clear intervals for scanning for jobs.
+let jobInterval;
+
+export const newActiveJob = (workflow, job) => ({
+    type: 'NEW_ACTIVE_JOB',
+    workflow,
     job
 });
 
-export const startJob = (job) => ({
-    type: 'START_JOB',
+export const jobCompleted = (job) => ({
+    type: 'JOB_COMPLETED',
     job
 });
 
+export const pollJobStatus = () => {
+    return (dispatch, getState) => {
+        const { connection: { uri, availableApis } } = getState();
+        const url = `http://${uri.split('/')[0]}${availableApis[0]}jobs`;
+        fetch(url, {
+            method: 'GET'
+        })
+        .then((response) => {
+            if (!response.ok) {
+                throw Error(response.statusText);
+            }
+            return response.json();
+        })
+        .then((json) => {
+            if (json.completed.length !== 0) {
+                json.completed.map(job => {
+                    if (Object.keys(job).length !== 0) {
+                        dispatch(actions.jobCompleted(job));
+                        dispatch(actions.refreshArtifacts());
```

Comment:
So do you mean putting the duplicates checking logic in the artifacts reducer? Then refresh could be removed entirely and just use load every time? I was trying to think of a way to handle the removal of jobs from the server dictionary, and think it could be done with a post/put after marking the job complete on the UI side. So like a,

```javascript
    .then((json) => {
        const completed = [];
        if (json.completed.length !== 0) {
            json.completed.map(job => {
                if (Object.keys(job).length !== 0) {
                    dispatch(actions.jobCompleted(job));
                    completed.push(job);
                }
            });
        }
        return completed;
    })
    .then((completed) => {
        fetch('/server/job/completed?', { method: POST?PUT?, headers ... })
    });